multi-node/generic/controller-install.sh

```diff
@@ -865,15 +771,20 @@ function start_addons {
     curl --silent -H "Content-Type: application/yaml" -XPOST -d"$(cat /srv/kubernetes/manifests/kube-dashboard-svc.yaml)" "http://127.0.0.1:8080/api/v1/namespaces/kube-system/services" > /dev/null
 }
 
-function enable_calico_policy {
+function start_calico {
+    # First we need to update the calico.yaml
+    /usr/bin/sed -i -e "s#\$ETCDENDPOINTS#$ETCD_ENDPOINTS#g" /srv/kubernetes/manifests/calico.yaml
+
     echo "Waiting for Kubernetes API..."
-    until curl --silent "http://127.0.0.1:8080/version"
+    # wait for the API
+    until curl --silent "http://127.0.0.1:8080/version/"
     do
         sleep 5
     done
     echo
-    echo "K8S: Calico Policy"
-    curl --silent -H "Content-Type: application/json" -XPOST -d"$(cat /srv/kubernetes/manifests/calico-system.json)" "http://127.0.0.1:8080/api/v1/namespaces/" > /dev/null
+
+    # Deploy Calico
+    docker run --rm --net=host -v /srv/kubernetes/manifests:/host/manifests $HYPERKUBE_IMAGE_REPO:$K8S_VER /hyperkube kubectl apply -f /host/manifests/calico.yaml
```

Comment:
This looks like it will get run on every boot - the kubectl create should fail "already exists", which is fine - but if the exit code propagates to the docker run, then this could cause the script to exit (I'm just not sure about the actual behavior -- ignore if this has been tested / is safe).